### PR TITLE
convenience functions

### PR DESCRIPTION
**Pull request by mountainMath**
Reworked descriptions right into `list_census_vectors`, shortened the `search_census_vectors` which closes https://github.com/mountainMath/cancensus/pull/58.

Also added two convenience functions `parent_census_vectors` and `child_census_vectors` to make it easier to select vectors through the hierarchy.

### CODE REVIEW

**Comment by mountainMath** [comment]
For the graph I would like to sort the categories by their overall prevalence (percentage, not total count) with the most prevalent first (at the bottom). Then I would like to sort the regions by the largest category. I don't know how to make this look pretty without pre-calculating these fields like I did earlier.

Maybe that's overkill for the vignette. Eventually I would like to encapsulate the code into a function where one specifies vectors (or maybe just a base vector), regions and a graph and axis titles and it spits out the graph. So one can easily run this for different census years. Or for different base variables, for example language spoken at home.

File: R/cancensus.R [hunk 1]
```diff
@@ -366,62 +378,107 @@ list_census_vectors <- function(dataset, use_cache=TRUE) {
...
 #' Query the CensusMapper API for vectors with descriptions matching a searchterm.
 #'
 #' @param searchterm The term to search for e.g. \code{"Ojibway"}.
 #' Search terms are case insensitive. If unable to find a given search term,
 #' this function will suggest the correct spelling to use when possible.
 #' @param dataset The dataset to query for available vectors, e.g.
 #'   \code{"CA16"}.
+#' @param type One of \code{NA}, \code{'Total'}, \code{'Male'} or \code{'Female'}.
+#' If specified, only return variables of specified `type`.
+#'   \code{"CA16"}.
```
Comment (atheriel) [inline R/cancensus.R]: This looks like a stray copy + paste.

File: R/cancensus.R [hunk 2]
```diff
@@ -366,62 +378,107 @@ list_census_vectors <- function(dataset, use_cache=TRUE) {
...
+#' @export
+#'
+#' @examples
+#' child_census_vectors(vector_list, 'CA16')
+child_census_vectors <- function(vector_list, leaves_only=FALSE){
+  dataset=attributes(base_list)$dataset
+  base_list <- vector_list
+  n=0
+  vector_list <- list_census_vectors(dataset) %>%
+    filter(parent_vector %in% base_list$vector) %>%
```
Comment (atheriel) [inline R/cancensus.R]: This `filter` (and many other functions in this PR) should be prefixed with `dplyr::`.

File: R/cancensus.R [hunk 3]
```diff
@@ -366,62 +378,107 @@ list_census_vectors <- function(dataset, use_cache=TRUE) {
...
-                par7list <- veclist[match(par6list$parent_vector, veclist$vector),]
-                result$parent7 <- par7list$label
-              }
-            }
-          }
-        }
-      }
-    }
+  result <- veclist[grep(searchterm, veclist$label, ignore.case = TRUE),]
+
```
Comment (atheriel) [inline R/cancensus.R]: Much nicer!

File: R/cancensus.R [hunk 4]
```diff
@@ -366,62 +378,107 @@ list_census_vectors <- function(dataset, use_cache=TRUE) {
...
+  # filter by type if needed
+  if (!is.na(type) && length(rownames(result)) > 0) {
+    result <- result[result$type==type,]
   }
+
   # Check if searchterm returned anything
-  if (length(rownames(result)) > 0 ) return(result)
+  if (length(rownames(result)) > 0 ) {
+    attributes(result)$dataset=dataset
```
Comment (atheriel) [inline R/cancensus.R]: `attr(result, "dataset") <- dataset` is generally preferred to `attributes(result)$dataset <- dataset` (here and elsewhere).

**Comment by mountainMath** [comment]
Thanks, made the changes.

**Pull request merged**